src/Dto/SearchDemand.php

```diff
@@ -4,15 +4,15 @@
 
 use Symfony\Component\HttpFoundation\Request;
 
-class SearchDemand
+readonly class SearchDemand
 {
-    public function __construct(protected string $query, protected int $page, protected array $filters)
+    public function __construct(private string $query, private int $page, private array $filters)
     {
     }
 
     public static function createFromRequest(Request $request): SearchDemand
     {
-        $requestFilters = $request->query->get('filters');
+        $requestFilters = $request->query->all()['filters'] ?? [];
```

Comment:
what's the difference here?

Reply:
The original code, when you selected some filter on the fronted (version or document type), produced an errorr: `Input value "filters" contains a non-scalar value.`